test/e2e/thv-operator/virtualmcp/helpers.go

```diff
@@ -694,6 +695,60 @@ func GetVMCPNodePort(
 	return nodePort
 }
 
+// WaitForVMCPHealthy polls the VirtualMCPServer's /health endpoint until it responds successfully.
+// This ensures the server is fully initialized and ready to handle MCP requests, which may happen
+// after the Kubernetes readiness probe passes but before the MCP server is fully initialized.
+func WaitForVMCPHealthy(nodePort int32, timeout time.Duration) {
```

Comment:
let's make the function accept a context so we can pass one from the parent for cancellation